catalogs/catalog-jdbc-doris/src/main/java/com/datastrato/gravitino/catalog/doris/operation/DorisTableOperations.java

```diff
@@ -194,6 +192,57 @@ static void appendIndexesSql(
     sqlBuilder.append(",").append(NEW_LINE).append(indexSql);
   }
 
+  private static void appendPartitionSql(
+      Transform[] partitioning, JdbcColumn[] columns, StringBuilder sqlBuilder) {
+    if (ArrayUtils.isEmpty(partitioning)) {
+      return;
+    }
+    Preconditions.checkArgument(
+        partitioning.length == 1, "Composite partition type is not supported");
+
+    StringBuilder partitionSqlBuilder = new StringBuilder();
+    Set<String> columnNames =
+        Arrays.stream(columns).map(JdbcColumn::name).collect(Collectors.toSet());
+
+    if (partitioning[0] instanceof Transforms.RangeTransform) {
+      partitionSqlBuilder.append(NEW_LINE).append(" PARTITION BY RANGE(");
+      // TODO support multi-column range partitioning in doris
+      Transforms.RangeTransform rangePartition = (Transforms.RangeTransform) partitioning[0];
+
+      Preconditions.checkArgument(
+          rangePartition.fieldName().length == 1, "Doris partition does not support nested field");
+      Preconditions.checkArgument(
+          columnNames.contains(rangePartition.fieldName()[0]),
+          "The partition field must be one of the columns");
+
+      String partitionColumn = BACK_QUOTE + rangePartition.fieldName()[0] + BACK_QUOTE;
+      // TODO we currently do not support pre-assign partition when creating range partitioning
```

Comment:
Pre-assign partition has been available since merge #3189. 

Would you like to implement it in this PR or a separate one? If not in this PR, kindly open a new issue for tracking this.

Reply:
Thanks for informing, I will do it in another PR, issue #3368 is created.

Reply:
If so, please modify the comment